Assets/Tests/PlayMode/Config/JsonConverters/ListOfStringsToEnumConverterTests.cs

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices.Tests.Config
+{
+    using Epic.OnlineServices.UI;
+    using NUnit.Framework;
+    using Newtonsoft.Json;
+    using System;
+    using System.Collections.Generic;
+    using Newtonsoft.Json.Linq;
+    public static partial class ListOfStringsToEnumConverterTests
+    {
+        private static readonly Dictionary<string, TestOrderEnum> CUSTOM_MAPPING = new()
+        {
+            { "FirstLetter", TestOrderEnum.A },
+            { "SecondLetter", TestOrderEnum.B },
+            { "ThirdLetter", TestOrderEnum.C },
+            { "AppleStartsWith", TestOrderEnum.A }
+        };
+
+        [Flags]
+        private enum TestOrderEnum : int
+        {
+            None = 0x00000,
+            A = 0x00001,
+            B = 0x00004,
```

Comment:
Should this be 0x00002? If tests are passing, I guess it doesn't matter much.

Reply:
It should not - because if it was, then the bitwise operators would not work they way we want them to. When the values are explicitly defined they need to be spaced such that the bitwise operators do not result in one of the enum values implying the other by default.

Reply:
That surprises me, given PlatformFlags being laid out like this:

```csharp
namespace Epic.OnlineServices.Platform
{
	[System.Flags]
	public enum PlatformFlags : ulong
	{
		None = 0x0,
		/// <summary>
		/// A bit that indicates the SDK is being loaded in a game editor, like Unity or UE4 Play-in-Editor
		/// </summary>
		LoadingInEditor = 0x00001,
		/// <summary>
		/// A bit that indicates the SDK should skip initialization of the overlay, which is used by the in-app purchase flow and social overlay. This bit is implied by <see cref="LoadingInEditor" />
		/// </summary>
		DisableOverlay = 0x00002,
		/// <summary>
		/// A bit that indicates the SDK should skip initialization of the social overlay, which provides an overlay UI for social features. This bit is implied by <see cref="LoadingInEditor" /> or <see cref="DisableOverlay" />
		/// </summary>
		DisableSocialOverlay = 0x00004,
		/// <summary>
		/// A reserved bit
		/// </summary>
		Reserved1 = 0x00008,
```

Here, it uses 0x00002 as the next value. Is this different, somehow?

Reply:
This is not different, it is intended functionality (see the comments on each successive value within the enum definition.

For the purposes of the _test_ I needed an example where this _wasn't_ the case.